rust/src/dataset.rs

```diff
@@ -338,7 +338,17 @@ impl Dataset {
         let dataset = if matches!(params.mode, WriteMode::Create) {
             None
         } else {
-            Some(Self::open(uri).await?)
+            // pull the store params from write params because there might be creds in there
+            Some(
+                Self::open_with_params(
+                    uri,
+                    &ReadParams {
+                        store_options: params.store_params.clone(),
+                        ..Default::default()
```

Comment:
Why not `params.clone()`?  Although the only other param I can see mattering on a write is perhaps `block_size`

Reply:
because the `params` here is `WriteParams`, what we want is a `ReadParams`.

I think this is a little bit of API defect here. Read path never writes, but write path often reads. I'll make a tickets for this